src/Compilers/CSharp/Portable/Binder/Binder_Constraints.cs

```diff
@@ -133,55 +135,80 @@ internal partial class Binder
                         {
                             var typeConstraintSyntax = (TypeConstraintSyntax)syntax;
                             var typeSyntax = typeConstraintSyntax.Type;
-                            if (typeSyntax.Kind() != SyntaxKind.PredefinedType && !SyntaxFacts.IsName(typeSyntax.Kind()))
+                            var typeSyntaxKind = typeSyntax.Kind();
+                            if (typeSyntaxKind != SyntaxKind.PredefinedType && typeSyntaxKind != SyntaxKind.PointerType && !SyntaxFacts.IsName(typeSyntax.Kind()))
                             {
                                 diagnostics.Add(ErrorCode.ERR_BadConstraintType, typeSyntax.GetLocation());
                             }
 
-                            var type = this.BindType(typeSyntax, diagnostics);
+                            var type = BindTypeOrUnmanagedKeyword(typeSyntax, diagnostics, out var isUnmanaged);
 
-                            // Only valid constraint types are included in ConstraintTypes
-                            // since, in general, it may be difficult to support all invalid types.
-                            // In the future, we may want to include some invalid types
-                            // though so the public binding API has the most information.
-                            if (!IsValidConstraintType(typeConstraintSyntax, type, diagnostics))
+                            if (isUnmanaged)
                             {
-                                continue;
-                            }
+                                if (constraints != 0 || constraintTypes.Any())
+                                {
+                                    diagnostics.Add(ErrorCode.ERR_UnmanagedConstraintMustBeAlone, typeSyntax.GetLocation());
```

Comment:
>diagnostics.Add(ErrorCode.ERR_UnmanagedConstraintMustBeAlone, typeSyntax.GetLocation()); [](start = 36, length = 88)

Why can't it be combined with an interface?

Reply:
It should be able to combine with interfaces. I think we discussed this on the tesplan review as a potential hole/followup issue.

---
In reply to: [172416658](https://github.com/dotnet/roslyn/pull/24817#discussion_r172416658) [](ancestors = 172416658)

Reply:
Vlad took care of that.

---
In reply to: [173018531](https://github.com/dotnet/roslyn/pull/24817#discussion_r173018531) [](ancestors = 173018531,172416658)